engines/alg/game_spacepirates.cpp

```diff
@@ -0,0 +1,1955 @@
+		return false;
+	}
+	uint32 header = inSaveFile->readUint32BE();
+	if (header != MKTAG('A', 'L', 'G', 'S')) {
+		warning("Unkown save file, header: %d", header);
+		return false;
+	}
+	inSaveFile->skip(1); // version, unused for now
+	_lives = inSaveFile->readSByte();
+	_shots = inSaveFile->readUint16LE();
+	_score = inSaveFile->readUint32LE();
+	_difficulty = inSaveFile->readByte();
+	uint16 sceneNum = inSaveFile->readUint16LE();
+	for (uint8 i = 0; i < 9; i++) {
+		_random_scenes_values[i] = inSaveFile->readUint16LE();
+		_random_scenes_used[i] = inSaveFile->readByte();
+	}
+	for (uint8 i = 0; i < 4; i++) {
+		_world_done[i] = inSaveFile->readByte();
+		_world_got_to[i] = inSaveFile->readUint16LE();
+	}
+	_max_random = inSaveFile->readByte();
+	_random_count_asteroids = inSaveFile->readByte();
+	_scene_before_flying_skulls = inSaveFile->readUint16LE();
+	_misc_rooms_count = inSaveFile->readByte();
+	_random_count_midship = inSaveFile->readByte();
+	_random_count = inSaveFile->readByte();
+	_got_to = inSaveFile->readUint16LE();
+	_player_died = inSaveFile->readByte();
+	_clue = inSaveFile->readUint16LE();
+	_shot_color = inSaveFile->readByte();
+	_shot_direction = inSaveFile->readByte();
+	_picked_start_splitter = inSaveFile->readUint16LE();
+	_current_world = inSaveFile->readSByte();
+	_selected_a_world = inSaveFile->readByte();
+	_selected_world_start = inSaveFile->readUint16LE();
+	_crystal_state = inSaveFile->readByte();
+	_shot_grin_reaper_count = inSaveFile->readByte();
+	_crystals_shot = inSaveFile->readByte();
+	_last_extra_life_score = inSaveFile->readUint32LE();
+	delete inSaveFile;
+	_game_loaded = true;
+	_gameInProgress = true;
+	_lives_loaded = _lives;
+	_shots_loaded = _shots;
+	_score_loaded = _score;
+	_difficulty_loaded = _difficulty;
+	_cur_scene = Common::String::format("scene%d", sceneNum);
+	_ChangeDifficulty(_difficulty);
+	if (!(_world_done[0] && _world_done[1] && _world_done[2] && _world_done[3]) && _current_world != -1) {
+		_cur_scene = "scene110";
+	}
+	return true;
+}
+
+// misc game functions
+void GameSpacePirates::_PlayErrorSound() {
+	_PlaySound(_skullSound);
+}
+
+void GameSpacePirates::_DisplayShotFiredImage() {
+	_DisplayShotFiredImage(&_mousePos);
+}
+
+void GameSpacePirates::_DisplayShotFiredImage(Common::Point *point) {
+	if (point->x >= _videoPosX && point->x <= (_videoPosX + _videoDecoder->getWidth()) && point->y >= _videoPosY && point->y <= (_videoPosY + _videoDecoder->getHeight())) {
+		uint16 targetX = point->x - _videoPosX - 4;
+		uint16 targetY = point->y - _videoPosY - 4;
+		AlgGraphics::drawImageCentered(_videoDecoder->getVideoFrame(), &_bulletholeIcon, targetX, targetY);
+	}
+}
+
+void GameSpacePirates::_DisplayShotLine(uint16 startX, uint16 startY, uint16 endX, uint16 endY) {
+	int16 currentX = startX;
+	int16 currentY = startY;
+	int16 deltaX = endX - startX;
+	int16 deltaY = endY - startY;
+	int16 stepX, stepY;
+	int16 steep, accum, error;
```

Comment:
`accum` and `error` are uninitialized, as the Windows builds show :)

```suggestion
	int16 steep, accum = 0, error = 0;
```

Reply:
On it. Weird that MinGW64/GCC did not complain about that.

Reply:
Should be fixed now I think